### PR TITLE
Port to Linter v1.

### PR DESCRIPTION
**Pull request by k2b6s9j**
Merge once all features all implemented and AtomLinter/linter-plus#24 is closed.

[Review on Reviewable](https://reviewable.io/reviews/atomlinter/linter-clang/83)

### CODE REVIEW

**Comment by k2b6s9j** [comment]
Implements #40 using the new API's Project View.

**Comment by k2b6s9j** [comment]
Fixes #40...

File: lib/linter-clang.coffee [hunk 1]
```diff
@@ -1,213 +1,100 @@
...
+        cmd = "#{cmd} #{Path.basename(filePath)}"
+        Data = []
+
+        Process = CP.exec(cmd, {cwd: Path.dirname(filePath)})
+        Process.stderr.on 'data', (data) -> Data.push(data.toString())
+        Process.on 'close', ->
+          Content = []
+          for line in Data
+            Content.push XRegExp.exec(line, regex)
+            console.log line# if atom.inDevMode()
```
Comment (steelbrain) [inline lib/linter-clang.coffee]: I think you missed this one

File: lib/linter-clang.coffee [hunk 2]
```diff
@@ -1,213 +1,87 @@
...
+    regex = ///
+      (.+): #File with issue
+      (\d+): #Line with issue
+      (\d+):\s #Column with issue
+      (.+):\s+ #Type of issue
+      (.*) #Message explaining issue
+    ///
+    return new Promise (Resolve) ->
+      filePath = TextEditor.getPath()
+      if filePath # Files that have not be saved
```
Comment (steelbrain) [inline lib/linter-clang.coffee]: This is redundant, https://github.com/AtomLinter/linter-ruby/pull/18/files#r33106620

**Comment by k2b6s9j** [comment]
Ready for code review.
/cc @steelbrain @AsaAyers @Snake231088 @m-bra

**Comment by steelbrain** [comment]
Glad to have you back my friend.

**Comment by k2b6s9j** [comment]
:smile:

**Comment by steelbrain** [comment]
The code LGTM, I am not in a position to test it any time soon though.

**Comment by k2b6s9j** [comment]
@steelbrain And I wouldn't force you to. Just as long you like the look of the code.

**Comment by steelbrain** [comment]
I do

**Comment by k2b6s9j** [comment]
Fixes #89.

Fixes #90.

**Pull request merged**